File: langs/qr.rb
# deps: gem install rqrcode chunky_png
require "rqrcode"
require "chunky_png"

qr = RQRCode::QRCode.new("https://example.com")
img = qr.as_png(size: 240)
IO.binwrite("qr.png", img.to_s)
puts "qr.png"

File: langs/qr.hs
-- deps: cabal install qrcode JuicyPixels
import Codec.QRCode
import Codec.Picture

main :: IO ()
main = do
  let code = encodeText (ErrorLevel M) Eight "https://example.com"
  let img = toImage 4 4 code
  writePng "qr.png" img
  putStrLn "qr.png"
